frontend/catalyst/jit.py

```diff
@@ -841,3 +862,18 @@ def compute_jvp(self, primals, tangents):
     @debug_logger
     def __call__(self, *args, **kwargs):
         return self.jaxed_function(*args, **kwargs)
+
+
+def erase_transform_primitives_from_jaxpr(jaxpr):
+    """
+    removes the `transform_named_sequence_p` and `apply_registered_pass_p`
+    primitives from a jaxpr
+    """
+
+    worklist = []
+    for primitive in jaxpr.eqns:
+        if primitive.primitive.name in ("transform_named_sequence", "apply_registered_pass"):
+            worklist.append(primitive)
+
+    for primitive in worklist:
+        jaxpr.eqns.remove(primitive)
```

Comment:
I would prefer keeping these in the JAXPR program.

Reply:
I have removed this erasing. The transform primitives will be kept in the jaxpr.